main.tf

```diff
@@ -299,8 +299,11 @@ resource "aws_lambda_event_source_mapping" "this" {
     for_each = try(each.value.filter_criteria, null) != null ? [true] : []
 
     content {
-      filter {
-        pattern = try(each.value["filter_criteria"].pattern, null)
+      dynamic "filter" {
+        for_each = try(each.value.filter_criteria, [])
```

Comment:
Please add newline after `for_each`.

Reply:
There should be a way to make it without introducing a break for existing users.

Please look at this PR where I did a similar change and added support for both types `map` and `list(map)` - https://github.com/terraform-aws-modules/terraform-aws-cloudfront/pull/91

Reply:
Tried to do this supporting both the object and list of object configuration. 

Should we use just the new notation in the examples? I kept one using the old notation for testing